src/main/java/net/neoforged/neoforge/common/extensions/ILevelReaderExtension.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.extensions;
+
+import java.util.Optional;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Holder;
+import net.minecraft.core.Registry;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.level.LevelReader;
+
+public interface ILevelReaderExtension {
+    default LevelReader self() {
+        return (LevelReader) this;
+    }
+
+    default boolean isAreaLoaded(BlockPos center, int range) {
```

Comment:
Moved over from the existing patch, as this can now live entirely in the interface.